kotlin/raster-rendering-rule/README.md

```diff
@@ -1,34 +1,36 @@
-
-This sample demonstrates how to create an `ImageServiceRaster`, fetch the `RenderingRules` from the service info, and use a `RenderingRule` to create an `ImageServiceRaster` and add it to a raster layer. 
-
-![Raster Rendering Rule App](raster-rendering-rule.png)
-
-## Features
-
-* ImageServiceRaster
-* RenderingRule
-* RasterLayer
-
-## Developer Pattern
-
- When the sample starts, an `ImageServiceRaster` is created and added to a `RasterLayer`.  The `RasterLayer` is then added to the map as an operational layer.  Connect to the `loadStatusChanged` signal for the `ImageServiceRaster`. Once the `ImageServiceRaster` is loaded, the `RenderingRuleInfos` are fetched. Iterate over each item in the `RenderingRuleInfos` to get the rendering rule name and populate `List` using the names. This becomes the list for the Android `Spinner`. When an item from the spinner is selected, the `RenderingRuleInfo` for the selected index is fetched from the service info. A `RenderingRule` object is created using the `RenderingRuleInfo` and applied to a newly created `ImageServiceRaster`. The `ImageServiceRaster` is then added to the `RasterLayer`.  
-
-```kotlin
-fun applyRenderingRule(imageServiceRaster: ImageServiceRaster, index: Int){
-    // clear all rasters
-    map.operationalLayers.clear()
-    // get the rendering rule info at the selected index
-    val renderRuleInfo = imageServiceRaster.serviceInfo.renderingRuleInfos[index]
-    // create a rendering rule object using the rendering rule info
-    val renderingRule = RenderingRule(renderRuleInfo)
-    // create a new image service raster
-    val appliedImageServiceRaster = ImageServiceRaster(resources.getString(R.string.image_service_url))
-    // apply the rendering rule
-    appliedImageServiceRaster.renderingRule = renderingRule
-    // create a raster layer using the image service raster
-    val rasterLayer = RasterLayer(appliedImageServiceRaster)
-    // add the raster layer to the map
-    map.operationalLayers.add(rasterLayer)
-}
-```
\ No newline at end of file
+# Raster rendering rule
+
+Display a raster on a map and apply different rendering rules to that raster.
+
+![Image of raster rendering rule](raster-rendering-rule.png)
+
+## Use case
+
+Raster images whose individual pixels represent elevation values can be rendered in a number of different ways, including representation of slope, aspect, hillshade, and shaded relief. Applying these different rendering rules to the same raster allows for a powerful visual analysis of the data. For example, a geologist could interrogate the raster image to map subtle geological features on a landscape, which may become apparent only through comparing the raster when rendered using several different rules.
+
+## How it works
+
+1.  Create an `ImageServiceRaster` using a URL to an online image service.
+2.  After loading the raster, use `imageServiceRaster.serviceInfo().renderingRuleInfos()` to get a list of `RenderingRuleInfo` supported by the service.
```

Comment:
Should these be class names rather than instance names?
```suggestion
2.  After loading the raster, use `ImageServiceRaster.serviceInfo().renderingRuleInfos()` to get a list of `RenderingRuleInfo` supported by the service.
```

Reply:
good question: usually when we've referred to the instantiation of a class in a previous step, then it's ok to use the informal variable call in this case. That's probably something we could be a bit more consistent about, I'll keep an eye out for that with future readmes too.

Reply:
Okay I'll stop correcting those then! 😄